src/devana/syntax_abstraction/functioninfo.py

```diff
@@ -290,11 +296,20 @@ def check_eq(a: FunctionInfo, b: FunctionInfo) -> bool:
     @property
     @lazy_invoke
     def modification(self) -> FunctionModification:
+        # for
         """Function modification enum flag."""
         self._modification = FunctionModification.NONE
-        for token in self._cursor.get_tokens():
+        tokens = list(self._cursor.get_tokens())
+        for i in range(len(tokens)):
+            token = tokens[i]
             if token.spelling == "constexpr":
                 self._modification |= FunctionModification.CONSTEXPR
+            if token.spelling == "consteval":
+                if i == 0:
+                    self._modification |= FunctionModification.CONSTEVAL
+                else:
+                    if tokens[i-1].spelling != "if":
+                        self._modification |= FunctionModification.CONSTEVAL
```

Comment:
It will be better to check that any { don't exist before consteval. This guarantees the absence of this keyword in the function body (note: check if it is not part of the initialization list)